pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1043,17 +1039,23 @@ func (vs *VSphere) DisksAreAttached(volPaths []string, nodeName k8stypes.NodeNam
 	_, vmDevices, dc, err := getVirtualMachineDevices(ctx, vs.cfg, vs.client, vSphereInstance)
 	if err != nil {
 		glog.Errorf("Failed to get VM devices for VM %#q. err: %s", vSphereInstance, err)
-		return attached, err
+		return nil, err
 	}
 
+	attached := make(map[string]bool)
 	for _, volPath := range volPaths {
-		result, _ := checkDiskAttached(volPath, vmDevices, dc, vs.client)
-		if result {
-			attached[volPath] = true
+		result, err := checkDiskAttached(volPath, vmDevices, dc, vs.client)
```

Comment:
Do we have similar concern when checkDiskAttached is called inside "DiskIsAttached" function?
Seems if err is not nil, it will always return false. And according to [the detach function here](https://github.com/kubernetes/kubernetes/blob/19795ea7c3d55f9eac453c212c74c537e4710902/pkg/volume/vsphere_volume/attacher.go#L240), even error is not nil, it will continue and mark this volume as already detached, which is also the same problem as resolved by this fix.